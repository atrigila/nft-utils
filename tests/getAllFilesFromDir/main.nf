workflow {

    def trace_timestamp = new java.util.Date().format( 'yyyy-MM-dd_HH-mm-ss')

    ch_stable_content = Channel.of(
    """
    I HAVE STABLE CONTENT
    """.stripIndent().trim())
    .collectFile(storeDir: "${params.outdir}/stable", name: 'stable_content.txt', sort: true, newLine: true)

    ch_stable_name = Channel.of(
    """
    I DO NOT HAVE STABLE CONTENT
    ${trace_timestamp}
    """.stripIndent().trim())
    .collectFile(storeDir: "${params.outdir}/stable", name: 'stable_name.txt', sort: true, newLine: true)

    ch_unstable_name = Channel.of(
    """
    I DO NOT HAVE STABLE NAME
    """.stripIndent().trim())
    .collectFile(storeDir: "${params.outdir}/not_stable", name: "${trace_timestamp}.txt", sort: true, newLine: true)
}
