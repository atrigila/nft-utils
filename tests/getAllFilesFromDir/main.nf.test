nextflow_pipeline {

    name "Test getAllFilesFromDir"
    script "./main.nf"
    tag "getAllFilesFromDir"

    test("getAllFilesFromDir") {
        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
          // Use getAllFilesFromDir() to get a list of all files and folders from the output directory, minus the timestamped files
          def stable_name    = getAllFilesFromDir(params.outdir, true, ['**/*[0-9]*.txt'] )
          // Use getAllFilesFromDir() to get a list of all files from the output directory, minus the non-stable files
          def stable_content = getAllFilesFromDir(params.outdir, false, 'tests/getAllFilesFromDir/.nftignore' )
          assert snapshot(
            // Only snapshot name
            stable_name*.name,
            // Snapshot content
            stable_content
          ).match()
        }
    }
}
