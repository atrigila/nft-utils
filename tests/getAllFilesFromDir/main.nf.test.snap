{
    "getAllFilesFromDir with named params": {
        "content": [
            [
                "pipeline_info",
                "stable",
                "stable_content.txt",
                "stable_name.txt"
            ],
            [
                "stable_content.txt:md5,f6d73f703cda1c725ea78369a6c3a48d"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.07.0"
        },
        "timestamp": "2024-10-02T10:15:47.491035"
    },
    "getAllFilesFromDir": {
        "content": [
            [
                "pipeline_info",
                "stable",
                "stable_content.txt",
                "stable_name.txt"
            ],
            [
                "stable_content.txt:md5,f6d73f703cda1c725ea78369a6c3a48d"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-30T13:09:08.845794"
    }
}
